main.py

```diff
@@ -16,106 +16,88 @@
 from email.mime.multipart import MIMEMultipart
 
 os.environ['PYTHONHTTPSVERIFY'] = '0'
+all_bugs = []
+all_tickets = []
 
 # function definitions
-def get_bugzilla(job_name):
+def get_bugzilla(job_name, bug_ids):
 
-	# get all bug ids for job from blocker file
-	try:
-		with open(blockers, 'r') as file:
-			bug_file = yaml.safe_load(file)
-			bug_ids = bug_file[job_name]['bz']
-	except Exception as e:
-		print("Error loading blocker configuration data (Bugzilla): ", e)
-		bugs = [{'bug_name': "Could not find relevant bug", 'bug_url': None}]
-	else:
+	# initialize bug list
+	bugs = []
 
-		# initialize bug list
-		bugs = []
+	# iterate through bug ids from blocker file
+	for bug_id in bug_ids:
 
-		# iterate through bug ids from blocker file
-		for bug_id in bug_ids:
+		# 0 should be default in YAML file (i.e. no bugs recorded)
+		# if there is a 0 entry then that should be the only "bug", so break
+		if bug_id == 0:
+			bugs = [{'bug_name': 'No bug on file', 'bug_url': None}]
+			break
 
-			# 0 should be default in YAML file (i.e. no bugs recorded)
-			# if there is a 0 entry then that should be the only "bug", so break
-			if bug_id == 0:
-				bugs = [{'bug_name': 'No bug on file', 'bug_url': None}]
-				break
+		# otherwise record real bug in overall list
+		all_bugs.append(bug_id)
 
-			# otherwise record real bug in overall list
-			all_bugs.append(bug_id)
+		# get bug info from bugzilla API
+		try:
 
-			# get bug info from bugzilla API
-			try:
+			# hotfix: API call does not work if '/' present at end of URL string
+			parsed_bz_url = config['bz_url'].rstrip('/')
 
-				# hotfix: API call does not work if '/' present at end of URL string
-				parsed_bz_url = config['bz_url'].rstrip('/')
-
-				bz_api = bugzilla.Bugzilla(parsed_bz_url)
-				bug = bz_api.getbug(bug_id)
-				bug_name = bug.summary
-			except Exception as e:
-				print("Bugzilla API Call Error: ", e)
-				bug_name = bug_id
-			finally:
-				bug_url = config['bz_url'] + "/show_bug.cgi?id=" + str(bug_id)
-				bugs.append(
-					{
-						'bug_name': bug_name, 
-						'bug_url': bug_url
-					}
-				)
+			bz_api = bugzilla.Bugzilla(parsed_bz_url)
+			bug = bz_api.getbug(bug_id)
+			bug_name = bug.summary
+		except Exception as e:
+			print("Bugzilla API Call Error: ", e)
+			bug_name = "BZ#" + str(bug_id)
+		finally:
+			bug_url = config['bz_url'] + "/show_bug.cgi?id=" + str(bug_id)
```

Comment:
I'd throw the `"/show_bug.cgi?id="` in a constant variable, and use the python url parsing library.

Reply:
there are more spots this could be applied too if you want too